flink-python/pyflink/testing/test_case_utils.py

```diff
@@ -90,3 +92,82 @@ def setUp(self):
         super(PyFlinkBatchTableTestCase, self).setUp()
         self.t_config = TableConfig.Builder().as_batch_execution().set_parallelism(4).build()
         self.t_env = TableEnvironment.get_table_environment(self.t_config)
+
+
+class PythonAPICompletenessTestCase(unittest.TestCase):
+    """
+    Base class for python api completeness tests, i.e.,
+    python api should be aligned with the Java API as much as possible.
+    """
+
+    @classmethod
+    def get_python_class_methods(cls, python_class):
+        return {cls.snake_to_camel(cls.java_method_name(method_name))
+                for method_name in dir(python_class) if not method_name.startswith('_')}
+
+    @staticmethod
+    def snake_to_camel(method_name):
+        output = ''.join(x.capitalize() or '_' for x in method_name.split('_'))
+        return output[0].lower() + output[1:]
+
+    @staticmethod
+    def get_java_class_method(java_class):
+        gateway = get_gateway()
+        s = set()
+        method_arr = gateway.jvm.Class.forName(java_class).getDeclaredMethods()
+        for i in range(0, len(method_arr)):
+            s.add(method_arr[i].getName())
+        return s
+
+    @classmethod
+    def check_methods(cls):
+        java_methods = PythonAPICompletenessTestCase.get_java_class_method(cls.java_class())
+        python_methods = cls.get_python_class_methods(cls.python_class())
+        missing_methods = java_methods - python_methods - cls.exclude_methods()
+        if len(missing_methods) > 0:
+            print(missing_methods)
+            print('The Exception should be raised after FLINK-12407 is merged.')
+            # raise Exception('Methods: %s in Java class %s have not been added in Python class %s.'
+            #                % (missing_methods, java_class, python_class))
+
+    @classmethod
+    def java_method_name(cls, python_method_name):
+        """
+        This method should be overWrite when the method name of the Python API cannot be
+        consistent with the Java API method name. we need to get the corresponding
```

Comment:
What about removing this line?
`we need to get the corresponding Java API method name based on the Python method name.`